pkg/envoy/lds/response.go

```diff
@@ -18,11 +18,6 @@ import (
 // 2. Outbound listener to handle outgoing traffic
 // 3. Prometheus listener for metrics
 func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_discovery.DiscoveryRequest, cfg configurator.Configurator, _ certificate.Manager, proxyRegistry *registry.ProxyRegistry) ([]types.Resource, error) {
-	if proxy.Kind() == envoy.KindGateway {
```

Comment:
I am inclined to build the listener for multicluster and ingress scenarios under this condition. I don't see a reason to build outbound listeners or prometheus (atm) for gateway, so let's bring this conditional back and create `lds/gateway.go` to house common code for ingress and multicluster (ex. listener struct), and simply return the LDS resources directly within this conditional.

Reply:
done!